app/service/file_svc.py

```diff
@@ -143,7 +145,18 @@ async def add_special_payload(self, name, func):
         :param func:
         :return:
         """
-        self.special_payloads[name] = func
+        if callable(func):
+            self.special_payloads[name] = func
+        else:
```

Comment:
since this is a big split here, let's add a comment as to what each block does. too small to split into separate functions.